arwen/host/directSimpleSC_test.go

```diff
@@ -0,0 +1,116 @@
+package host
+
+import (
+	"errors"
+	"math/big"
+
+	mock "github.com/ElrondNetwork/arwen-wasm-vm/mock/context"
+	"github.com/stretchr/testify/require"
+)
+
+func wasteGasChildMock(instanceMock *mock.InstanceMock, config interface{}) {
+	testConfig := config.(directCallGasTestConfig)
+	instanceMock.AddMockMethod("wasteGas", simpleWasteGasMockMethod(instanceMock, testConfig.gasUsedByChild))
+}
+
+func failChildMock(instanceMock *mock.InstanceMock, config interface{}) {
+	instanceMock.AddMockMethod("fail", func() *mock.InstanceMock {
+		host := instanceMock.Host
+		instance := mock.GetMockInstance(host)
+		host.Runtime().FailExecution(errors.New("forced fail"))
+		return instance
+	})
+}
+
+func execOnSameCtxParentMock(instanceMock *mock.InstanceMock, config interface{}) {
+	testConfig := config.(directCallGasTestConfig)
+	instanceMock.AddMockMethod("execOnSameCtx", func() *mock.InstanceMock {
+		host := instanceMock.Host
+		instance := mock.GetMockInstance(host)
+		t := instance.T
+		host.Metering().UseGas(testConfig.gasUsedByParent)
+
+		arguments := host.Runtime().Arguments()
+		input := DefaultTestContractCallInput()
+		input.GasProvided = testConfig.gasProvidedToChild
+		input.CallerAddr = instance.Address
+		input.RecipientAddr = arguments[0]
+		input.Function = string(arguments[1])
+		numCalls := big.NewInt(0).SetBytes(arguments[2]).Uint64()
+
+		for i := uint64(0); i < numCalls; i++ {
```

Comment:
Optional: rewrite this loop to behave like the one in "execOnDestCtx" below, iterating over triples from the `arguments` array.

Reply:
done